### PR TITLE
Moved syslog from 2/ to 2and3/.  Allow both forms of syslog.syslog.

### PR DESCRIPTION
**Pull request by matt-kimball**
The syslog module still exists in Python 3.  Moved it from `stdlib/2/`
to `stdlib/2and3/`.

Also, the syslog.syslog implementation accepts two forms of arguments:

```
   syslog(priority: int, message: str)
   syslog(message: str)
```

The previous stub only allowed the first of the two forms.  Now
either are accepted.

### CODE REVIEW

File: stdlib/2and3/syslog.pyi [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from typing import (Optional, Union)
```
Comment (gvanrossum) [inline stdlib/2and3/syslog.pyi]: This should not have parentheses.

File: stdlib/2and3/syslog.pyi [hunk 2]
```diff
@@ -35,4 +37,4 @@ def LOG_UPTO(a: int) -> int: ...
 def closelog() -> None: ...
 def openlog(ident: str = ..., logoption: int = ..., facility: int = ...) -> None: ...
 def setlogmask(x: int) -> int: ...
-def syslog(priority: int, message: str) -> None: ...
+def syslog(priority: Union[int, str], message: str = ...) -> None: ...
```
Comment (gvanrossum) [inline stdlib/2and3/syslog.pyi]: Actually I just realized this is still not correct. It should be an overload for syslog(int, str) and syslog(str). (The current version also allows syslog(str, str), which is incorrect.)

Comment (matt-kimball) [inline stdlib/2and3/syslog.pyi]: Fixed to use overload now.

**Comment by gvanrossum** [comment]
Yeah!

**Pull request merged**